control-plane-operator/hostedclusterconfigoperator/cmd.go

```diff
@@ -210,6 +215,14 @@ func (o *HostedClusterConfigOperator) Run(ctx context.Context) error {
 	if err := mgr.Add(cpCluster); err != nil {
 		return fmt.Errorf("cannot add CPCluster to manager: %v", err)
 	}
+	var kubevirtInfraConfig *rest.Config
+	if _, err := os.Stat(o.InfraKubeconfig); err == nil {
+		kubevirtInfraConfig = operator.CfgFromFile(o.InfraKubeconfig)
+	} else {
+		// in case infra kubeconfig hasn't been provided, default the kubevirtInfraCluster to cpConfig
+		kubevirtInfraConfig = cpConfig
+	}
+
```

Comment:
this should probably read slightly differently.

as it's written right now, by not inspecting if InfraKubeconfig is set when a stat error potentially occurs, we might default to the cpConfig even when o.InfraKubeconfig is set.

I think something like this resolves the potential issue. 
```
if o.InfraKubeconfig != "" {
                kubevirtInfraConfig = operator.CfgFromFile(o.InfraKubeconfig)
} else {
		// in case infra kubeconfig hasn't been provided, default the kubevirtInfraCluster to cpConfig
		kubevirtInfraConfig = cpConfig
}

```

Reply:
the idea was to verify the kubeconfig file actually exists in the file system before attempting to construct a client from it.
anyway, i'm changing as per your suggestion.